### PR TITLE
MQE-1124: [Github Issue] Allow running tests for modules installed in…

### PR DESCRIPTION
**Pull request by aljcalandra**
… Magento instance via vendor directory. magento/magento2-functional-testing-framework#162

- Adjusted Module Resolver to look Recursively from Project Root, Magento BP and dev/tests
- Updated Module Resolver Unit Tests to reflect changes

Adjusted ModuleResolver to do a recursive search in directories related to PROJECT_ROOT and MAGENTO_BP

### Fixed Issues (if relevant)

1. magento/magento2-functional-testing-framework#<162>: Allow running tests for modules installed in Magento instance via vendor directory.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

**Comment by aljcalandra** [comment]
magento/magento2-functional-testing-framework/#162

File: src/Magento/FunctionalTestingFramework/Util/ModuleResolver.php [hunk 1]
```diff
@@ -265,7 +270,8 @@ private function globRelevantPaths($testPath, $pattern)
         }
 
         foreach ($relevantPaths as $codePath) {
-            $mainModName = basename(str_replace($pattern, '', $codePath));
+            $allComponents = $this->getRegisteredModuleList();
```
Comment (KevinBKozan) [inline src/Magento/FunctionalTestingFramework/Util/ModuleResolver.php]: Need to do two things here:
* Strip `/Test/Mftf` off end of codepath
* Assign to $modulePaths according to $allComponents value match (Magento_Store -> Store) to match module list

File: src/Magento/FunctionalTestingFramework/Util/ModuleResolver.php [hunk 2]
```diff
@@ -28,6 +28,16 @@ class ModuleResolver
      */
     const CUSTOM_MODULE_PATHS = 'CUSTOM_MODULE_PATHS';
 
+    /**
+     * List of path types present in Magento Component Registrar
+     */
+    const PATHS = ['module', 'library', 'theme', 'language', 'setup'];
```
Comment (KevinBKozan) [inline src/Magento/FunctionalTestingFramework/Util/ModuleResolver.php]: Need to remove `setup` from PATHS.

File: src/Magento/FunctionalTestingFramework/Util/ModuleResolver.php [hunk 3]
```diff
@@ -28,6 +28,16 @@ class ModuleResolver
...
+
+    /**
+     * Magento Registrar Class
+     */
+    const REGISTRAR_CLASS = "\Magento\Framework\Component\Registrar";
```
Comment (KevinBKozan) [inline src/Magento/FunctionalTestingFramework/Util/ModuleResolver.php]: Class should be `\Magento\Framework\Component\ComponentRegistrar`

**Comment by KevinBKozan** [comment]
@aljcalandra I'm going to triple check the last bit of changes against a git-pull installation of Magento, I'll let you know the results.

**Comment by aljcalandra** [comment]
Resolved magento/magento2-functional-testing-framework/issues/162

**Pull request merged**